debug_toolbar/templates/debug_toolbar/stacktrace.html

```diff
@@ -0,0 +1,3 @@
+{% for s in stacktrace %}<span class="djdt-path">{{s.0}}</span><span class="djdt-file">{{s.1}}</span> in <span class="djdt-func">{{s.3}}</span>(<span class="djdt-lineno">{{s.2}}</span>)
+    <span class="djdt-code">{{s.4}}</span>
+{% if show_locals %}<pre class="djdt-locals">{{s.5|pprint}}</pre>{% endif %}{% endfor %}
```

Comment:
the problem with that markup is that all of this it's rendered inside a <pre>, and I had to format it that without newlines so that no newlines were added in the rendered stacktrace in the places no newlines should not be rendered. I will think of a better logic for it

Reply:
Nope, what you have is probably best. I tried to use `{% spaceless %}` in the template, but it ended up just as ugly.